### ISSUE TITLE
Disabling a Gem from the project through the Project Manager and rebuilding it does not remove that Gem from enabled_gems.cmake file

### ISSUE THREAD

**Issue by LB-ArturZieba** [o3de/o3de#14222]
**Describe the bug**
Removing Gems that are written to `<project>\Gem\enabled_gems.cmake` via Project Manager, does not remove them from that file, causing them to remain functional in the Editor.

Please note that the issue only occurs for the gems that were manually enabled in the project through the Project Manager earlier.

It should also be noted that deleting Cache and rerunning Asset Processor after disabling a Gem does not fix this issue. However removing the Gem's entry from the `<project>\Gem\enabled_gems.cmake` fixes this issue.

**Prerequisites**
* Make sure the used project is using the Default Template for easier verification.
* Make sure that the Gem used in **Steps to reproduce** is actually written to the `<project>\Gem\enabled_gems.cmake` file. Some of the Gems - for example *Game State Samples* or *Primitive Assets* Gems are not written to that file at all, while *AWS GameLift*, *Achievements*, *Barrier Input* or *PhysX Debug* are.

**Steps to reproduce**
1. In the Project Manager click on the project's hamburger button and select *Configure Gems...* from the dropdown.
2. Enable a Gem in the project (for example the *PhysX Debug* Gem).
3. Click the Save button in the bottom right corner.
4. Notice that the Gem was added to the enabled_gems.cmake file (in <project_location>/Gem folder).
5. In the Project Manager click on the project's hamburger button and select *Configure Gems...* from the dropdown.
6. Disable the gem enabled in step 2 (in this case the *PhysX Debug* gem).
7. Click the Save button in the bottom right corner.
8. Verify the content of the enabled_gems.cmake file.

**Expected behavior**
Gem is removed from the enabled_gems.cmake file.

**Actual behavior**
Gem remains in the enabled_gems.cmake file, causing it to remain functional in the Editor.

**Video**

https://user-images.githubusercontent.com/86952082/214843206-9d3f021f-0b82-4815-a5e9-33893c95227d.mp4

**Found in Branch**
Development (57a500f)

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
57a500f3eaf2b0d8450003dd04016f5147e940a7

**Desktop**
 - Device: PC
 - OS: Windows
 - Version 11
 - CPU AMD Ryzen 5 3600
 - GPU NVIDIA GeForce RTX 2060 SUPER
 - Memory 16GB

**Comment by lumberyard-employee-dm** [comment]
@amzn-phist @AMZN-alexpete Do you know if this is caused by any of the changes made for supporting optional gem references and adding semantic versioning support to the project.json recently?

**Comment by lumberyard-employee-dm** [comment]
This issue doesn't reproduce with using the CLI o3de.py script
```powershell
PS D:\o3de> scripts\o3de.bat disable-gem -pp AutomatedTesting -gn PhysXDebug
```

Assigning the issue to SIG-Content

**Comment by AMZN-alexpete** [comment]
likely my changes, taking a look.

**Comment by amzn-phist** [comment]
This issue sounds similar to #12328

**Comment by LB-KacperMetelski** [comment]
Issue has been verified as fixed in commit 9291f40.
